modules/olmv1-catalog-plain.adoc

```diff
@@ -0,0 +1,234 @@
+
+. Push the image to your remote registry:
++
+[source,terminal]
+----
+$ podman push quay.io/<organization_name>/<repository_name>:<image_tag>
+----
+
+.Additional resources
+
+* link:https://github.com/operator-framework/olm-docs/blob/master/content/en/docs/Reference/file-based-catalogs.md[File-based catalog bundle schema]
+* link:https://github.com/opencontainers/image-spec#oci-image-format-specification[OCI image specification]
+* link:https://github.com/operator-framework/rukpak/blob/main/docs/bundles/plain.md#image-source[RukPak > Building a plain bundle > Image source]
+* link:https://github.com/operator-framework/rukpak/blob/main/docs/sources/image.md#private-image-registries[RukPak > Sources > Images > Private image registries]
+
+[id="olmv1-creating-fbc_{context}"]
+= Creating a file-based catalog
+
+If you do not have a file-based catalog, you must perform the following steps to initialize the catalog.
+
+.Procedure
+
+. Create a directory for the catalog by running the following command:
++
+[source,terminal]
+----
+$ mkdir <catalog_dir>
+----
+
+. In the same directory level, create a Dockerfile that can build a catalog image:
++
+[NOTE]
+====
+The Dockerfile must be in the same parent directory as the catalog directory that you created in the previous step:
+
+.Example directory structure
+[source,terminal]
+----
+.
+├── <catalog_dir>
+└── <catalog_dir>.Dockerfile
+----
+====
++
+.Example `<catalog_dir>.Dockerfile`
+[source,yaml]
+----
+FROM scratch <1>
+ADD <catalog_dir> /configs
+----
+<1> Use the `FROM scratch` directive to make the size of the image smaller. No other files or directories are required in the bundle image.
+
+. Populate the catalog with the package definition for your extension by running the `opm init` command:
++
+[source,terminal]
+----
+$ opm init <extension_name> \
+    --output json \
+    > <catalog_dir>/index.json
+----
++
+This command generates an `olm.package` declarative config blob in the specified catalog configuration file.
+
+[id="olmv1-adding-plain-bundle-to-fbc_{context}"]
+= Adding a plain bundle to a file-based catalog
+
+Currently, the `opm render` command does not support adding plain bundles to catalogs. You must manually add plain bundles to your file-based catalog, as shown in the following procedure.
+
+.Procedure
+
+. Verify that your catalog's `index.json` or `index.yaml` file is similar to the following example:
++
+.Example `<catalog_dir>/index.json` file
+[source,json]
+----
+{
+    {
+     "schema": "olm.package",
+     "name": "<extension_name>",
+    }
+}
+----
+
+. To create an `olm.bundle` blob, edit your `index.json` or `index.yaml` file, similar to the following example:
++
+.Example `<catalog_dir>/index.json` file with `olm.bundle` blob
+[source,json]
+----
+{
+   "schema": "olm.bundle",
+    "name": "<extension_name>.v<version>",
+    "package": "<extension_name>",
+    "image": "quay.io/<organization_name>/<repository_name>:<image_tag>",
+    "properties": [
```

Comment:
It's better to add the `olm.gvk` and `olm.bundle.object` types.

Reply:
@jianzhangbjz why do we have to add those types? My assumption was that for plain bundle, they're not really required.